src/data/checkfiles/default_checkfiles/unused-metadata.js

```diff
@@ -0,0 +1,148 @@
+
+    let specificQuery =
+      "SELECT count(t.*) FROM (select meta_id from R_META_MAIN except select meta_id from R_OBJT_METAMAP) t;";
+
+    let specificWarning = false;
+    const specificResp = await axios({
+      url: `${this.restApiLocation}/query`,
+      method: "GET",
+      headers: {
+        Authorization: localStorage.getItem("zmt-token"),
+      },
+      params: {
+        query: specificQuery,
+        type: "specific",
+        limit: 999999999,
+        offset: 0,
+        "case-sensitive": 1,
+      },
+    }).catch((e) => {
+        specificWarning = true;
+    });
+
+    if (specificResp && specificResp.data) {
+      numUnusedMeta = specificResp.data._embedded[0][0];
+    } else {
+      let totalMetaQuery = "SELECT META_DATA_ATTR_ID";
+
+      const totalMetaResp = await axios({
+        url: `${this.restApiLocation}/query`,
+        method: "GET",
+        headers: {
+          Authorization: localStorage.getItem("zmt-token"),
+        },
+        params: {
+          query: totalMetaQuery,
+          type: "general",
+          limit: 999999999,
+          offset: 0,
+          "case-sensitive": 0,
+        },
+      }).catch((e) => {});
+
+      if (totalMetaResp && totalMetaResp.data) {
+        totalMetaResp.data._embedded.forEach((row) => {
+          totalMeta.add(row[0]);
+        });
+      }
+
+      console.log("totalMeta:", totalMeta);
+
+      const getMeta = async (query) => {
+        const resp = await axios({
+          url: `${this.restApiLocation}/query`,
+          method: "GET",
+          headers: {
+            Authorization: localStorage.getItem("zmt-token"),
+          },
+          params: {
+            query: query,
+            type: "general",
+            limit: 999999999,
+            offset: 0,
+            "case-sensitive": 0,
+          },
+        }).catch((e) => {
+          console.log(e);
+        });
+
+        let useSet = new Set();
+        if (resp && resp.data) {
+          console.log("resp.data", resp.data);
+
+          // turn the first item in each row into a set
+          resp.data._embedded.forEach((row) => {
+            useSet.add(row[0]);
+          });
+        }
+
+        return useSet;
+      };
+
+      // find all metadata associated with users
+      let userMetaQuery = "SELECT META_USER_ATTR_ID, USER_ID";
+      let userMetaSet = await getMeta(userMetaQuery);
+
+      let dataObjMetaQuery = "SELECT META_DATA_ATTR_ID, DATA_ID";
+      let dataObjMetaSet = await getMeta(dataObjMetaQuery);
+
+        let rescMetaQuery = "SELECT META_RESC_ATTR_ID, RESC_ID";
+        let rescMetaSet = await getMeta(rescMetaQuery);
+
+        let collMetaQuery = "SELECT META_COLL_ATTR_ID, COLL_ID";
+        let collMetaSet = await getMeta(collMetaQuery);
+
+        numUnusedMeta = totalMeta.size - userMetaSet.size - dataObjMetaSet.size - rescMetaSet.size - collMetaSet.size;
+        console.log("numUnusedMeta", numUnusedMeta);
```

Comment:
indentation.    and needs some additional comments to be consistent (or a single comment for this section should talk about all the different 'nouns')